### ISSUE TITLE
Fix type of LabelClasses.classes

### ISSUE THREAD

**Issue by lossyrob** [stac-utils/pystac#618]
Currently setting a LabelClasses.classes property to List[str] gives an error in Pylance, e.g.

```python
classes: List[str] = ...
label = LabelExtension.ext(item, add_if_missing=True)
label.apply(
    label_description="USGS GAP/LANDFIRE",
    label_type=LabelType.RASTER,
    label_classes=[
        LabelClasses.create(
            classes=classes,
            name="",
        )
    ],
)
```

yields error at `classes=classes` with

```
Argument of type "list[str]" cannot be assigned to parameter "classes" of type "List[str | int | float]" in function "create"
  TypeVar "_T@list" is invariant
    Type "str" cannot be assigned to type "str | int | float"
      "str" is incompatible with "int"
      "str" is incompatible with "float"
```
This I believe is due to the Union type as the generic to List [here](https://github.com/stac-utils/pystac/blob/69e9f76b40ed6399b78c671636311e42f2cd17be/pystac/extensions/label.py#L76) and [here](https://github.com/stac-utils/pystac/blob/69e9f76b40ed6399b78c671636311e42f2cd17be/pystac/extensions/label.py#L109) 

This should be `Union[List[str], List[int], List[float]]` instead. 

Bonus: search for other instances of `List[Union[...]]` and convert to `Union[List[.], List[.], ...]` to avoid a similar issue if it exists elsewhere.

**Comment by cuttlefish** [comment]
Is there a reason to use `List` over `Sequence` here? It probably doesn't make sense to take mutable data types as inputs?
I believe that `Sequence` can maintain the heterogeneity of the datatype (which may not be desired in this specific case, but would be useful elsewhere).
I've got a branch with the `LabelExtension` converted to use `Sequence` if there's any interest. This would probably only make sense if we were intending to transition everything over to `Sequence`.

**Comment by duckontheweb** [comment]
Closed via #627
